src/main/scala/io/qbeast/spark/delta/IndexStatusBuilder.scala

```diff
@@ -68,24 +69,29 @@ private[delta] class IndexStatusBuilder(
     val revisionAddFiles = revisionFiles
     import revisionAddFiles.sparkSession.implicits._
     val blocks = revisionAddFiles
-      .flatMap(IndexFiles.fromAddFile(root.dimensionCount)(_).blocks)
-      .collect()
+      .map(IndexFiles.fromAddFile(root.dimensionCount))
+      .flatMap(_.blocks)
+      .as[Block]
+      .toLocalIterator()
+      .asScala
       .toIndexedSeq
+
```

Comment:
How does this function work? Why are we putting every blocks in the root node, this is not correct, right?